core/logging.py

```diff
@@ -52,27 +46,95 @@ def emit(self, record: logging.LogRecord):
         )
 
 
-oldLevels = {k: v.getEffectiveLevel()
-             for k, v in logging.root.manager.loggerDict.items()
-             if isinstance(v, logging.Logger)}
+# ViURLocalFormatter ---------------------------------------------------------------------------------------------------
+
+class ViURLocalFormatter(logging.Formatter):
+    """
+    This is a formatter that injects console color sequences for debug output.
+    """
+
+    BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE = range(8)
+    RESET_SEQ = "\033[0m"
+    COLOR_SEQ = "\033[1;%dm"
+    BOLD_SEQ = "\033[1m"
+
+    LEVELS = {
+        "WARNING": YELLOW,
+        "INFO": CYAN,
+        "DEBUG": BLUE,
+        "CRITICAL": RED,
+        "ERROR": RED
+    }
+
+    def format(self, record):
+        if "pathname" in record.__dict__.keys():
+            # truncate the pathname
+            if "/deploy" in record.pathname:
+                pathname = record.pathname.split("/deploy/")[1]
+            else:
+                pathname = record.pathname
+                if len(pathname) > 20:
+                    pathname = ".../" + "/".join(pathname.split("/")[-3:])
```

Comment:
Imagine you have a module in a very depth package like: `google.cloud.resourcemanager_v3.services.projects.client`. 
You would get `.../services/projects/client`, for which it is not quite clear where it belongs. 
If we remove the middle part, you can see the root package (_google_) too: `google/.../projects/client`

```suggestion
                    parts = pathname.split("/")
                    del parts[1:-3]
                    parts.insert(1, "...")
                    pathname = "/".join(parts)
```

Reply:
That's a nice solution!